integrations/kubecost/integration.py

```diff
@@ -0,0 +1,156 @@
+        # Regular expression to validate the format of the date pair value
+        regex = r"^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z,\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z$"
+        if not re.match(regex, value):
+            raise ValueError(
+                "Invalid date pair format. Use 'YYYY-MM-DDTHH:MM:SSZ,YYYY-MM-DDTHH:MM:SSZ'."
+            )
+
+
+class UnixtimePairField(str):
+    @classmethod
+    def validate(cls, value: str) -> None:
+        # Regular expression to validate the format of the unixtime pair value
+        regex = r"^\d+,\d+$"
+        if not re.match(regex, value):
+            raise ValueError(
+                "Invalid unixtime pair format. Use 'X,Y' (X and Y are positive integers)."
+            )
+
+
+class AggregationField(str):
+    @classmethod
+    def validate(cls, value: str) -> None:
+        # Regular expression to validate the format of the aggregation value
+        regex = r"^(cluster|node|namespace|controllerKind|controller|service|pod|container|invoiceEntityID|accountID|provider|label:name|annotation:name)(,(cluster|node|namespace|controllerKind|controller|service|pod|container|invoiceEntityID|accountID|provider|label:name|annotation:name))*$"
+        if not re.match(regex, value):
+            raise ValueError(
+                "Invalid aggregation format. Use 'cluster', 'node', 'namespace', 'controllerKind', 'controller', 'service', 'pod', 'container', 'label:name', or 'annotation:name'."
+            )
+
+
+class DurationField(str):
+    @classmethod
+    def validate(cls, value: str) -> None:
+        # Regular expression to validate the format of the step value
+        regex = r"^\d+[smhd]$"
+        if not re.match(regex, value):
+            raise ValueError(
+                "Invalid duration format. Use 'Xs', 'Xm', 'Xh', or 'Xd' (X is a positive integer)."
+            )
+
+
+class KubecostSelector(Selector):
+    window: Literal[
+        "today",
+        "week",
+        "month",
+        "yesterday",
+        "lastweek",
+        "lastmonth",
+        "30m",
+        "12h",
+        "7d",
+    ] | DatePairField | UnixtimePairField = Field(default="today")
+    aggregate: AggregationField | None = Field(
+        description="Field by which to aggregate the results.",
+    )
+    step: DurationField | None = Field(
+        description="Duration of a single allocation set (e.g., '30m', '2h', '1d'). Default is window.",
+    )
+    accumulate: bool = Field(
+        default=False,
+        description="If true, sum the entire range of sets into a single set. Default value is false",
+    )
+    idle: bool = Field(
+        default=True,
+        description="If true, include idle cost (i.e. the cost of the un-allocated assets) as its own allocation",
+    )
+    external: bool = Field(
+        default=False,
+        description="If true, include external, or out-of-cluster costs in each allocation. Default is false.",
+    )
+    filterClusters: str | None = Field(
```

Comment:
pls when writing python stick with snake case
for fields that has camel case you can use the alias attribute to set it to expect something else
for example here itll be 
```
    filter_clusters: str | None = Field(alias="filterClusters", ....
```